csharp/src/Ice/IObjectPrx.cs

```diff
@@ -128,19 +127,20 @@ public void IcePing(IReadOnlyDictionary<string, string>? context = null)
         public Task IcePingAsync(IReadOnlyDictionary<string, string>? context = null,
                                  IProgress<bool>? progress = null,
                                  CancellationToken cancel = default)
-            => IcePingAsync(context, progress, cancel, synchronous: false);
-
-        private Task IcePingAsync(IReadOnlyDictionary<string, string>? context, IProgress<bool>? progress,
-                                  CancellationToken cancel, bool synchronous)
         {
-            var request = OutgoingRequestFrame.Empty(this, "ice_ping", idempotent: true, context);
-            var task = IceInvokeAsync(request, oneway: IsOneway, progress, cancel, synchronous);
-            return IsOneway ? task : AwaitResponseAsync(task);
-
-            static async Task AwaitResponseAsync(Task<IncomingResponseFrame> task)
-                => (await task.ConfigureAwait(false)).ReadVoidReturnValue();
+            var request = IceI_IcePingCreateRequest(context);
+            var task = IceInvokeAsync(request, IsOneway, progress, cancel);
+            return IsOneway ? task : IceAwaitResponse(task, IceI_IcePingReadResponse);
         }
 
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        private OutgoingRequestFrame IceI_IcePingCreateRequest(IReadOnlyDictionary<string, string>? context)
```

Comment:
Should we provide a single protected helper method for all requests with an empty payload, and a single protected helper method for all requests with a void response?

Reply:
I favor using the same style as non-void functions